docs/contributors/git-workflow.md

```diff
@@ -1,26 +1,94 @@
+
+## Overview
+
+An overview of the process for contributors is:
+
 - Fork the Gutenberg repository.
 - Clone the forked repository.
 - Create a new branch.
 - Make code changes.
+- Confirm tests pass.
 - Commit the code changes within the newly created branch.
 - Push the branch to the forked repository.
-- Submit a Pull Request to the Gutenberg repository.
+- Submit a pull request to the Gutenberg repository.
+
+See the [repository management document](/docs/contributors/repository-management.md) for additional information on how the Gutenberg project uses GitHub.
+
+## Git Workflow Walkthrough
+
+The workflow for code and documentation is the same, since both are managed in GitHub. You can watch a [video walk-through of contributing documentation](https://wordpress.tv/2020/09/02/marcus-kazmierczak-contribute-developer-documentation-to-gutenberg/) and the accompanying [slides for contributing to Gutenberg](https://mkaz.blog/wordpress/contribute-documentation-to-gutenberg/).
+
+
+Here is a visual overview of the Git workflow:
+
+![Visual Overview of Git Workflow](https://developer.wordpress.org/files/2020/09/git-workflow.png)
+
+
+**Step 1**: Go to the Gutenberg repository on GitHub and click Fork. This creates a copy of the main Gutenberg repository to your account.
+
+![Screenshot showing fork button on GitHub](https://developer.wordpress.org/files/2020/09/gutenberg-fork.png)
+
+**Step 2**: Clone your forked repository locally, it is located at: `https://github.com/YOUR-USER-NAME/gutenberg`. Cloning copies all the files to your computer. Open a terminal and run:
+
+```bash
+git clone https://github.com/YOUR-USER-NAME/gutenberg
+```
+
+This will create a directory called `gutenberg` with all the files for the project. It might take a couple of minutes, it is downloading the entire history of the Gutenberg project.
+
+**Step 3**: Create a branch for your change, see below for branch naming. For this example the branch name is the complete string: `update/my-branch`
+
+```bash
+git switch -c update/my-branch
+```
+
+**Step 4**: Make the code changes. Build, confirm, and test your change thoroughly. See [coding guidelines](/docs/contributors/coding-guidelines.md) and [testing overview](/docs/contributors/testing-overview.md) for guidance.
+
+**Step 5**: Commit your change with a [good commmit message](https://make.wordpress.org/core/handbook/best-practices/commit-messages/). This will commit your change to your local copy of the repository.
+
+```bash
+git commit -m "Your Good Commit Message" path/to/FILE
+```
+
+**Step 6**: Push your change up to GitHub. The change will be pushed to your copy of the repository on the GitHub
+
+```bash
+git push -u origin upgrade/my-branch
+```
+
+**Step 7**: Go to your repository on GitHub, it will automatically detect the change and give you a link to create a pull request.
```

Comment:
```suggestion
**Step 7**: Go to your forked repository on GitHub -- it will automatically detect the change and give you a link to create a pull request.
```